parlai/tasks/dbll_babi/agents.py

```diff
@@ -55,8 +55,113 @@ def __init__(self, opt, shared=None):
         opt['datafile'] = _path(os.path.join('babi', 'babi1'), params, opt)
         opt['cands_datafile'] = _path(os.path.join('babi', 'babi1'), params,
                                       opt, 'train')
+        self.opt = opt
         super().__init__(opt, shared)
 
+    def setup_data(self, path):
+        if len(self.opt['task'].split(':')) == 4:
+            return self.setup_data_with_feedback(path)
+        else:
+            return super().setup_data(path)
+
+    def setup_data_with_feedback(self, path):
```

Comment:
@jaseweston what do you think of this input setup?

Reply:
len(self.opt['task'].split(':')) == 4 should more specifically search for a string name and have a comment at least

Reply:
I meant the actual data:
that is, you feel good about appending the next x to the current x?

Reply:
(I'm understanding that correctly, right?

that the data looks like:
```
Mary moved to the bedroom.
Mary travelled to the garden.
Where is John?
No, that's wrong.
[labels: garden]
```

Reply:
Your understanding is correct.